tests/test_templates.py

```diff
@@ -87,116 +107,151 @@
-                                          [[4.83], [1.70]],
-                                          [[4.74], [5.39]],
-                                          [[0.88, 0.62], [1.09, 3.02]],
-                                          [[-0.01, -0.05], [0.08, -0.19]],
-                                          [[1.89, 3.59], [1.49, 3.71]],
-                                          [[0.09, 0.03], [-0.14, 0.04]]
-                                          ], {}),
-                     (Interferometer, [[2.31], [3.49], [0.98, 1.54]], {})
-                     ]
+CV_DIFFABLE_NONDIFFABLE = [(DisplacementEmbedding,
+                            {'features': [1., 2.]},
+                            {}),
+                           (SqueezingEmbedding,
+                            {'features': [1., 2.]},
+                            {}),
+                           (CVNeuralNetLayers,
+                            {'theta_1': [[2.31], [1.22]],
+                             'phi_1': [[3.47], [2.01]],
+                             'varphi_1': [[0.93, 1.58], [5.07, 4.82]],
+                             'r': [[0.21, 0.12], [-0.09, 0.04]],
+                             'phi_r': [[4.76, 6.08], [6.09, 6.22]],
+                             'theta_2': [[4.83], [1.70]],
+                             'phi_2': [[4.74], [5.39]],
+                             'varphi_2': [[0.88, 0.62], [1.09, 3.02]],
+                             'a': [[-0.01, -0.05], [0.08, -0.19]],
+                             'phi_a': [[1.89, 3.59], [1.49, 3.71]],
+                             'k': [[0.09, 0.03], [-0.14, 0.04]]},
+                            {}),
+                           (Interferometer,
+                            {'theta': [2.31],
+                             'phi': [3.49],
+                             'varphi': [0.98, 1.54]},
+                            {})
+                           ]
 
 
 #########################################
 # Circuits shared by test classes
 
 
-def qnode_qubit_args(dev, intrfc, templ1, templ2, n, hyperp1, hyperp2):
-    """QNode for qubit integration circuit using positional arguments"""
-    hyperp1['wires'] = range(2)
-    hyperp2['wires'] = range(2)
+def qnode_qubit_args(dev, interface, template1, template2, n_args1):
+    """Qubit qnode factory passing differentiable parameters as positional arguments"""
 
-    @qml.qnode(dev, interface=intrfc)
-    def circuit(*inp):
-        # Split inputs again
-        inp1 = inp[:n]
-        inp2 = inp[n:]
-        # Circuit
-        qml.PauliX(wires=0)
-        templ1(*inp1, **hyperp1)
-        templ2(*inp2, **hyperp2)
-        qml.PauliX(wires=1)
-        return [qml.expval(qml.Identity(0)), qml.expval(qml.PauliX(1))]
+    # Signature juggling to pass diffable as single positional arg, but keep track of input names
+    @qml.qnode(dev, interface=interface)
+    def circuit(*diffable, keys_diffable1=None, keys_diffable2=None, nondiffable1=None, nondiffable2=None):
```

Comment:
I'm a bit confused as to what these different arguments should correspond to?

Reply:
Updated the comment a bit. All that clutter just there to pass the values of the ``diffable`` dictionary, but keep track of the keys to turn it back into a dictionary.